### ISSUE TITLE
parse_method, first argument can be a list

### ISSUE THREAD

**Issue by wvxvw** [Parquery/swagger-to#68]
This line has an error: https://github.com/Parquery/swagger-to/blob/master/swagger_to/swagger.py#L284

`raw_dict` can be a list of multiple `RawDict`.

**Comment by mristin** [comment]
Hi @wvxvw ,
Thanks a lot for spotting the error. Could you maybe give me a couple of more details? Could you provide a short swagger example?

**Comment by wvxvw** [comment]
Hey, now this makes me wonder, is version 3 (the one that's now officially called Open API) supposed to be supported? This could be the culprit. The one I worked with was a v3.

**Comment by mristin** [comment]
No, it's OpenAPI v2. V3 was still not available/well-established when we started working on swagger-to.

**Comment by mristin** [comment]
Hi @wvxvw ,
Can we close the issue?

**Comment by abingham** [comment]
I think I'm running into this issue as well:
```
(acquisition-system)  % python `which swagger_to_elm_client.py` --swagger_path swagger.yaml --outdir gen                                                                                                         static/elm (apispec ⚡) opabinia
Traceback (most recent call last):
  File "/Users/abingham/.virtualenvs/acquisition-system/bin/swagger_to_elm_client.py", line 73, in <module>
    main()
  File "/Users/abingham/.virtualenvs/acquisition-system/bin/swagger_to_elm_client.py", line 44, in main
    swagger, errs = swagger_to.swagger.parse_yaml_file(path=swagger_path)
  File "/Users/abingham/.virtualenvs/acquisition-system/lib/python3.8/site-packages/swagger_to/swagger.py", line 431, in parse_yaml_file
    return parse_yaml(stream=fid)
  File "/Users/abingham/.virtualenvs/acquisition-system/lib/python3.8/site-packages/swagger_to/swagger.py", line 393, in parse_yaml
    path, path_errors = _parse_path(raw_dict=path_dict)
  File "/Users/abingham/.virtualenvs/acquisition-system/lib/python3.8/site-packages/swagger_to/swagger.py", line 332, in _parse_path
    method, method_errors = _parse_method(raw_dict=method_dict)
  File "/Users/abingham/.virtualenvs/acquisition-system/lib/python3.8/site-packages/swagger_to/swagger.py", line 286, in _parse_method
    adict = raw_dict.adict
AttributeError: 'list' object has no attribute 'adict'
```
Is this a problem with my swagger yaml (produced by flask-restx) or with swagger_to? Would it help if I posted the yaml?

**Comment by mristin** [comment]
Hi @abingham,
Let me have a closer look and I'll come back to you.

**Comment by mristin** [comment]
Hi @abingham ,
Could you please post a short specs that breaks swagger-to so that I can debug (and include it in the tests)?

I'll add a validation step using the JSON schema for **OpenAPI 2** (https://github.com/OAI/OpenAPI-Specification/blob/master/schemas/v2.0/schema.json).

**Comment by abingham** [comment]
Here's a short spec that passes other validators I've tried but which breaks swagger-to:
```
basePath: /api/v1
consumes:
- application/json
info:
  description: An API
  title: An API
  version: '1.0'
paths:
  /foo/{foo_id}:
    get:
      operationId: get_foo
      responses:
        '200':
          description: Success
      tags:
      - foo
    parameters:
    - in: path
      description: The foo id
      name: foo_id
      required: true
      type: string
produces:
- application/json
swagger: '2.0'
tags:
- description: description
  name: foo 

```

**Comment by mristin** [comment]
@abingham thanks! At the first sight the schema looks good, so it is indeed a bug in swagger-to. I'll try to investigate the issue till this weekend.

**Comment by mristin** [comment]
Hi @abingham ,
The bug was indeed in swagger-to. I was ignorant of common parameters (see https://swagger.io/docs/specification/2-0/describing-parameters/, Section "Common Parameters"). I made a fix in PR #103.

Could you please check that the HEAD of the master branch resolves your issue and that it works on your whole schema?

(Btw., if you haven't done so, may I kindly ask you to star the project on GitHub to give it some visibility?)

**Comment by abingham** [comment]
This bug seems to be fixed now. Thanks!

I've found another issue which I'll report separately.
